test/utils_test.js

```diff
@@ -431,6 +432,60 @@ describe("utils", () => {
       expect(mergeObjects(obj1, obj2)).eql(expected);
     });
 
+    describe("trimObject()", () => {
+      it("should't mutate the provided objects", () => {
+        const obj = { a: undefined };
+        trimObject(obj);
+        expect(obj).eql({ a: undefined });
+      });
+
+      it("should trim empty object", () => {
+        const obj = {};
+        const res = trimObject(obj);
+        expect(res).eql(undefined);
+      });
+
+      it("should trim object with no fields set", () => {
+        const obj = { a: undefined, b: undefined, c: undefined };
+        const res = trimObject(obj);
+        expect(res).eql(undefined);
+      });
+
+      it("shouldn't trim object with fields set", () => {
+        const obj = { a: true, b: undefined, c: undefined };
+        const res = trimObject(obj);
+        expect(res).eql({ a: true, b: undefined, c: undefined });
+      });
+
+      it("should trim field object", () => {
+        const obj = { a: true, o: { b: undefined } };
+        const res = trimObject(obj);
+        expect(res).eql({ a: true, o: undefined });
```

Comment:
This might be a bit pedantic, but why return an object with `undefined` s in it, and not e.g. `{a: true}`? (Of course, this would require changes to your implementation.